api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2591,6 +2598,132 @@ public void deleteObjectTags(DeleteObjectTagsArgs args)
     executeDelete(args, null, queryParams);
   }
 
+  /**
+   * Uploads multiple objects in a single put call. It is done by creating intermediate TAR file
+   * optionally compressed which is uploaded to S3 service.
+   *
+   * <pre>Example:{@code
+   * // Upload snowball objects.
+   * List<SnowballObject> objects = new ArrayList<SnowballObject>();
+   * objects.add(
+   *     new SnowballObject(
+   *         "my-object-one",
+   *         new ByteArrayInputStream("hello".getBytes(StandardCharsets.UTF_8)),
```

Comment:
Having to provide all streams upfront makes it seem un-scalable, either having everything in memory or as FDs seem like a big limitation.

Would it be possible to do it similar to the go version where you can send them one at the time?

Reply:
`SnowballObject` class accepts one of the options
1. `InputStream` (like `io.Reader` in golang), size and optional modification date
2. just a filename

In the example, I just used a bytes as `InputStream` for simplicity.